src/queueManager.js

```diff
@@ -220,7 +228,73 @@ export default class QueueManager {
       }
     });
   }
-  _saveEvents() {
+    // unregister token for existing user.
+    _unregisterTokenApiCall(unregisterData,callback) {
+
+        // this._uploading = true; Do i need this for Kaios ?
+
+        var url = this._getEndPoint();
+        url = this._addToURL(url, 'sn', Utils.getNow()); // send epoch seconds as seq number
+        url = this._addARPToRequest(url);
+        url = this._addToURL(url, "r", new Date().getTime()); // add epoch millis to beat caching of the URL
+
+        let meta = {
+            id: Account.getAccountId(),
+            "SDK Version": version,
+            s: `${Session.getSessionId()}`,
+        };
+
+        if (Account.getAppVersion()) {
+            meta.Version = `${Account.getAppVersion()}`;
+        }
+
+        var guid = data.guid;
+        if (guid) {
+            url = this._addToURL(url, "gc", guid);
+            meta.g = guid;
+        }
+
+        Utils.log.debug(`kaios vapid req with request url : ${url} and meta: ${meta}`);
+
+        url = this._addToURL(url, "d", Helpers.compressData(JSON.stringify(meta)));
+
+
+        // Utils.log.debug(`Sending kaios Vapid request: ${JSON.stringify()}`);  TODO
+
+        var _this = this;
+        var data = [];
+        data.push(unregisterData);
+        new Request(url,data).send( function(status, response) {
+            // _this._uploading = false;
+            response = response || {};
+            Utils.log.debug(`kaios req handling response with status: ${status} and data: ${JSON.stringify(response)}`);
+
+            try {
+                if (status === 200) {
+
+                        //Device.setGUID(response.KVAPID);
+                        if (typeof callback === 'function') {
+                            callback(); // call callback to register subscription to kaios servers.
+                        }
+
+                    // handle errors
+                    // request too large
+                } else if (status === 413) {
+                    Utils.log.error('413 on un-register Token request');
+                } else {
+                    Utils.log.error(`kaios vapid request failed with status ${status}.  Will retry.`);
```

Comment:
There is no code to retry here. Update the message or add the code.

Reply:
updated